### PR TITLE
allow ozaria to play cpp levels

### PR DESCRIPTION
**Pull request by smallst**
fixes ENG-4

these levels require update (mainly on 2 `for` with same variable like `int i`, need to update to single declearation. or checking `&&` `||` these logical operator for goals). capstone levels may work. just verifier reports it has issue. but manually test works good
```
CH3.Capstone.b
CH3.M4.L3.L2.B
CH3.M4.L3.L1.B
CH3.M4.L2.CL1.B
CH3.M4.L1.L3.B
CH3.M3.L1.L1.B
CH3.M2.L2.L5.B
CH3.M2.L2.L4.B 
CH3.M2.L2.L3.B 
CH3.M1.L1.L4.B
1UP.M4.L2.L2.B
1UP.M4.L2.L3.B
1UP.M4.L2.L4.B
CH4.M1.L3.L2
CH4.M1.L3.L3
CH4.M1.L4.L2
CH4.M1.L4.L3
CH4.M1.L4.L4
CH4.M1.L4.L5
CH4.M2.L1.L1
CH4.M2.L1.L2
CH4.M2.L1.L3
CH4.M3.L2.CL1
CH4.Capstone.EX1
CH4.Capstone.EX2
CH4.Capstone.EX3
CH4.Capstone
```

### CODE REVIEW

File: ozaria/site/views/play/level/tome/Spell.coffee [hunk 1]
```diff
@@ -60,6 +61,8 @@ module.exports = class Spell
   setLanguage: (@language) ->
     @language = 'html' if @level.isType('web-dev')
     @displayCodeLanguage = utils.capitalLanguages[@language]
+    if @language in ['cpp', 'java', 'lua', 'coffeescript', 'python'] and not @languages[@language]
```
Comment (mrfinch) [inline ozaria/site/views/play/level/tome/Spell.coffee:64]: why add all languages? Or we need java, lua too?

File: ozaria/site/views/play/level/tome/SpellView.coffee [hunk 2]
```diff
@@ -833,49 +840,73 @@ module.exports = class SpellView extends CocoView
...
   # - Problem alerts and ranges will only show on fully cast worlds. Annotations will show continually.
 
+  fetchToken: (source, language) =>
+    if language not in ['java', 'cpp']
+      return Promise.resolve(source)
+    else if source of @loadedToken
+      return Promise.resolve(@loadedToken[source])
+
+    headers =  { 'Accept': 'application/json', 'Content-Type': 'application/json' }
+    m = document.cookie.match(/JWT=([a-zA-Z0-9.]+)/)
```
Comment (mrfinch) [inline ozaria/site/views/play/level/tome/SpellView.coffee:850]: are we using this variable?

File: ozaria/site/views/play/level/tome/SpellView.coffee [hunk 3]
```diff
@@ -833,49 +840,73 @@ module.exports = class SpellView extends CocoView
...
+    service = window?.localStorage?.kodeKeeperService or "https://asm14w94nk.execute-api.us-east-1.amazonaws.com/service/parse-code-kodekeeper"
+    if me.useChinaServices()
+      headers['Authorization'] = 'APPCODE b3e285d032a343db8bd2b51a05a5ff1d'
```
Comment (mrfinch) [inline ozaria/site/views/play/level/tome/SpellView.coffee:853]: ok with exposing this in public?

File: ozaria/site/views/play/level/tome/Spell.coffee (hunk 1 above)
Comment (smallst) [reply ozaria/site/views/play/level/tome/Spell.coffee:64]: i just copy from coco. maybe we can enable other langs in ozaria later too ? i think it is not blocking anyway

File: ozaria/site/views/play/level/tome/SpellView.coffee (hunk 3 above)
Comment (smallst) [reply ozaria/site/views/play/level/tome/SpellView.coffee:853]: 🤔 it is public in china branch for a long time, but maybe not in master ? i can clean this code

File: ozaria/site/views/play/level/tome/SpellView.coffee (hunk 2 above)
Comment (smallst) [reply ozaria/site/views/play/level/tome/SpellView.coffee:850]: interesting. seems we don't. let me double check it and clean it

**Review by mrfinch — APPROVED** [review]

**Pull request merged**